### ISSUE TITLE
RETAINED message state red text not shown for new messages

### ISSUE THREAD

**Issue by mars000** [thomasnordquist/MQTT-Explorer#662]
running windows 0.4.0 beta 1 on windows 11 (latest build)
when new retained messages received does the app does not show the usual "REATINED x" text in app window.
I disconnect and reconnect client and then it does show

**Comment by yawor** [comment]
This is correct according to MQTT protocol. Clients only get retained bit set in the message if the message has been published as retained before the client connected. Otherwise the client doesn't get this information.

The solution would be for the client to act as an MQTT Bridge, because bridges get all the info from the upstream broker. But AFAIK before MQTTv5, bridges where not formalised and brokers had their own custom implementations for creating bridges (for example a mosquitto broker could only act as a bridge for another mosquitto broker). In MQTTv5 there's a native support for bridges, but MQTT Explorer doesn't support this version of protocol.

**Comment by ejohb** [comment]
Not the fault of ME of course, but this same issue just cost me about and hour of confusion and debugging. I spotted the (very small) "Retained messages only appear to be retained when client connects after the initial publish." tool tip over the "retain" checkbox, but though it was referring to _other_ clients.

IMO, this message should be

a) Much more prominent, perhaps even a pop-up with "Don't show this again" option. And
b) Clearer, e.g. "Retained messages only appear to be retained when MQTT Explorer connects after the initial publish."

**Comment by fabian727** [comment]
same here. 
Stumbled upon this "problem" for some hours now, before starting googling.

As every component is working as specified, maybe a small message or so would be nice (for beginners)
